reactor-core/src/main/java/reactor/util/context/Context.java

```diff
@@ -191,32 +190,21 @@ static Context of(Map<?, ?> map) {
 	}
 
 	/**
-	 * Resolve a value given a key that exists within the {@link Context}, or throw
-	 * a {@link NoSuchElementException} if the key is not present.
+	 * Create a {@link Context} out of a {@link ContextView}, enabling write API on top of
+	 * the read-only view. If the {@link ContextView} is already a {@link Context}, return
+	 * the same instance.
 	 *
-	 * @param key a lookup key to resolve the value within the context
-	 * @param <T> an unchecked casted generic for fluent typing convenience
-	 *
-	 * @return the value resolved for this key (throws if key not found)
-	 * @throws NoSuchElementException when the given key is not present
-	 * @see #getOrDefault(Object, Object)
-	 * @see #getOrEmpty(Object)
-	 * @see #hasKey(Object)
+	 * @param contextView the {@link ContextView} to convert (or cast) to {@link Context}
+	 * @return the converted {@link Context} for further modifications
 	 */
-	<T> T get(Object key);
+	static Context fromReadOnly(ContextView contextView) {
```

Comment:
Perhaps we should not add such method. One can still do the `empty().putAll()` trick, but adding the method would help with the following wrong pattern that we should discourage:
```java
Mono.deferWithContext(ctx -> {
    Context.fromReadOnly(ctx).put("I have", "no idea what I am doing");
    return Mono.just("foo");
});

Reply:
having the casting trick helps avoiding a bit of junk. I don't think `fromReadOnly` is misleading compared to plain exposition of full `Context` API (which _guides_ you towards shooting yourself in the foot).

maybe I should rename it to `fromView(ContextView)` ? or even `copyView`? (even though for a `ContextView` that is also a `Context`, this is not strictly true).

Reply:
WDYT about... `Context.of(ContextView)`? We have these factory methods already, and it would hint that a new context is created

Reply:
yeah I was using `of` in my initial local work but somehow felt less explicit. that said, indeed all `of` methods are obviously about creating new contexts so that could work.